src/test/java/org/jenkins/plugins/lockableresources/DeclarativePipelineTest.java

```diff
@@ -0,0 +1,116 @@
+package org.jenkins.plugins.lockableresources;
+
+import com.google.common.base.Joiner;
+import hudson.model.Result;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.assertNull;
+
+public class DeclarativePipelineTest {
+  @Rule
+  public JenkinsRule j = new JenkinsRule();
+
+  @Test
+  public void basicTest() throws Exception {
+    WorkflowJob p = j.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+      new CpsFlowDefinition(m("pipeline {",
+        " agent none",
+        " options {",
+        "  lock resource: 'resource1'",
```

Comment:
No, it does not.
"lock" should have 0 arguments but has 1 arguments instead. @ line 4, column 3.
... will check

Reply:
In that case it breaks old (and still valid) scenario.

Reply:
Yes, unfortunately. At least to my limited knowledge of the Steps API, this means that this PR won't work :(

Because a property is 'required' if (and only if) there is no associated setter method.
And that simplified specification (lock 'resource') apparently only works if that one parameter is a required one.

So, we can't have both: if it is required, we can use the simple form, but need to specify the resource in any case. If we want to allow resource being optional (and in fact it is), that simple form no longer works.

Reply:
> Yes, unfortunately. At least to my limited knowledge of the Steps API, this means that this PR won't work :(
> 
> Because a property is 'required' if (and only if) there is no associated setter method. And that simplified specification (lock 'resource') apparently only works if that one parameter is a required one.
> 
> So, we can't have both: if it is required, we can use the simple form, but need to specify the resource in any case. If we want to allow resource being optional (and in fact it is), that simple form no longer works.

This is probably the reason, why I have documented it as a limitation.
Please try to ask the community (https://gitter.im/jenkinsci/jenkins). Maybe is there some solution / workaround. I do not know. When we have no solution, I will propose to keep the changes in test-code. It is still good idea to test also declarative pipeline.
The other idea is to create new step lockResource(Map options). But to eliminate many duplicated code here, I will propose to create shared-library (groovy code) as e extension to this plugin. When we want it, please join to our gitter channel https://gitter.im/jenkinsci/lockable-resources and we can discuss it there or better here https://github.com/jenkinsci/lockable-resources-plugin/discussions/445.
Maybe we shall organize a short call.

BR Martin